src/software/simulation/physics/physics_ball.cpp

```diff
@@ -133,13 +175,12 @@ bool PhysicsBall::isInFlight()
             !isTouchingOtherObject())
         {
             in_flight_origin = std::nullopt;
-            return false;
-        }
-        else
-        {
-            return true;
         }
     }
+}
 
-    return false;
+bool PhysicsBall::isInFlight() const
+{
+    updateIsInFlight();
```

Comment:
I feel like the 'updateIsInFlight()' doesn't fit here.

I think the users assumption would be that this function returns a state variable, not doing updating.

Is this required to make the check work? If this isn't run how does the simulator know that the ball is in the air?

Reply:
@garethellis0 ?

Reply:
This is technically not required as most of the time this function is called during the PhysicsWorld stepSimulation function.

Reply:
I think from the perspective of the user it makes sense. I expect `isInFlight` to do whatever work it needs to in order to figure out if the ball is in flight or not. 

One suggestion here might be to make `updateisInFlight` a `protected` member so it's more clear to the user that they're not supposed to touch it?

Reply:
Update from Gareth: "This actually doesn't really need to be protected. The interface is the same for the user regardless, since they can't tell there are side effects. Eg calling `isInFlight()` before or after `updateIsInFlight()`, or twice in a row, won't yield different results. Since it's consistent from an interface POV, it should be fine as is"

Reply:
@EvanMorcom ?

Reply:
@EvanMorcom

Reply:
We should have a quick in-person discussion about this today

Reply:
@EvanMorcom @garethellis0 I need resolution on this please